src/js/relay.firefox.com/get_profile_data.js

```diff
@@ -1,419 +1,453 @@
 (async function () {
-  // Get the api token from the account profile page
-  const profileMainElement = document.querySelector("#profile-main");
-  const apiToken = profileMainElement.dataset.apiToken;
-  browser.storage.local.set({ apiToken });
-
-  // API URL is ${RELAY_SITE_ORIGIN}/api/v1/
-  const { relayApiSource } = await browser.storage.local.get("relayApiSource");
-
-  const apiProfileURL = `${relayApiSource}/profiles/`;
-  const apiRelayAddressesURL = `${relayApiSource}/relayaddresses/`;
-
-  async function apiRequest(url, method = "GET", body = null, opts=null) {
-
-    const cookieString =
-      typeof document.cookie === "string" ? document.cookie : "";
-    const cookieStringArray = cookieString
-      .split(";")
-      .map((individualCookieString) => individualCookieString.split("="))
-      .map(([cookieKey, cookieValue]) => [
-        cookieKey.trim(),
-        cookieValue.trim(),
-      ]);
-
-    const [_csrfCookieKey, csrfCookieValue] = cookieStringArray.find(
-      ([cookieKey, _cookieValue]) => cookieKey === "csrftoken"
-    );
+  const dahsboardInitializationObserver = new MutationObserver((mutations) => {
+    mutations.forEach(mutation => {
+      if (mutation.type !== "childList") {
+        return;
+      }
+      mutation.addedNodes.forEach(node => {
+        if (node.id !== "profile-main") {
+          return;
+        }
 
-    browser.storage.local.set({ csrfCookieValue: csrfCookieValue });
+        // Once an element `#profile-main` is added, the dashboard is initialized
+        // with the data needed by the add-on; stop watching for futher changes,
+        // and run once:
+        dahsboardInitializationObserver.disconnect();
+        run();
+      });
+    });
+  });
+  if (document.getElementById("profile-main") === null) {
+    dahsboardInitializationObserver.observe(document.body, { childList: true, subtree: true });
+  } else {
+    run();
+  }
```

Comment:
This diff looks intimidating, but the above is the only code that's new. Everything below is exactly the same as it was, except for being wrapped inside the `run()` function.